multibody/kinematic/kinematic_evaluator_set.h

```diff
@@ -57,6 +57,50 @@ class KinematicEvaluatorSet {
   /// Again, note that this is an index set into the other object, not self.
   std::vector<int> FindUnion(KinematicEvaluatorSet<T> other);
 
+  /// Compute M(q) * d/dt v, given the state, control inputs and constraint
+  /// forces. Forces are associated with the full kinematic elements.
+  /// @param context
+  /// @param lambda constraint forces, applied via
+  ///   evaluators.EvalActiveJacobian().transpose() * lambda
+  drake::VectorX<T> CalcMassMatrixTimesVDot(
+      const drake::systems::Context<T>& context,
+      const drake::VectorX<T>& lambda) const;
+
+  /// Computes vdot given the state, control inputs and constraint
+  /// forces. Similar to CalcMassMatrixTimesVDot, but uses inv(M)
+  /// and includes qdot. Forces are associated with the full kinematic
+  /// elements.
+  /// @param context
+  /// @param lambda constraint forces, applied via
+  ///   evaluators.EvalActiveJacobian().transpose() * lambda
+  drake::VectorX<T> CalcTimeDerivativesGivenForce(
```

Comment:
GivenForce sounds weird. Depending on which function you anticipate being called more often, it might make more sense to rename the function with the user supplied lambda CalcTimeDerivatives(), and rename the current CalcTimeDerivatives() to be CalcTimeDerivativesAndForces()

Reading it a second time, it makes more sense. Make rename it to CalcTimeDerivativesWithForce() or CalcTimeDerivativesUsingForce()?

Reply:
Done.